src/jade/tabs/contacts.jade

```diff
@@ -99,7 +99,7 @@ section.col-xs-12.content(ng-controller="ContactsCtrl")
       span.name(ng-hide='editing', ng-bind="entry.name") 
     .col-xs-12.col-sm-6.col-md-6(ng-hide='editing')
       div
-        b.hidden-xs {{entry.address | rpaddressorigin}}: {{entry.address}}
+        b.hidden-xs {{entry.address | rpaddressorigin}}: {{contactNames[entry.address] || resolveName(entry.address)}}
```

Comment:
I c, LGTM (Though, I just saw there is ```rpripplename``` in filters.filter.js. Which means ... you could possibly reduce the commit to ```entry.address | rpripplename:{tilde:true}```)